### DISCUSSION TITLE
Error loading workspace folders (expected 1, got 0) failed to load view for ...: exhausted

### DISCUSSION THREAD

**Discussion by mrkam2** [golang/vscode-go#2207] [Help]
How do I debug/troubleshoot this "exhausted" error? I've tried to apply all the suggestions mentioned and linked [here](https://github.com/golang/vscode-go/discussions/1365) but it didn't work. Is there a better way to troubleshoot it?

**Comment by hyangah** [comment]
how large is the workspace? It looks like the error message indicates the workspace is too big for a gopls to handle (1M+ files) at once. The suggestion is to set up a directory filter to limit your workspace assuming you don't need all those 1M+ files.

**Reply by mrkam2** [reply]
> how large is the workspace? It looks like the error message indicates the workspace is too big for a gopls to handle (1M+ files) at once. The suggestion is to set up a directory filter to limit your workspace assuming you don't need all those 1M+ files.

The workplace is pretty big and besides go projects it includes projects in other languages, some are Node JS with their extensive `node_modules` subfolders. For example, one such projects has 373K files in its `node_modules` directory. 

Can the plugin more actively filter out files that are irrelevant to the go project?

**Reply by hyangah** [reply]
That's why the filtering setting exists. By default, the top-level `node_modules` are excluded from scanning, but you can tell the extension so even it can skip scanning to see if there is a go code hidden in deeply nested directory.

OTOH I also wish gopls stops scanning the entire workspace - I hope work for https://github.com/golang/go/issues/53278 brings improvements to this problem domain. Users still need to use `go.work` to scope their workspace.

**Reply by mrkam2** [reply]
> filtering setting exists

What filtering setting are you referring to?

**Reply by mrkam2** [reply]
I guess you're referring to https://github.com/golang/tools/blob/master/gopls/doc/settings.md#directoryfilters-string.

**Reply by mrkam2** [reply]
There isn't a way to use a mask to specify multiple folders in one entry, is it? @hyangah
